portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -1606,9 +1607,20 @@ private void _checkFileEntry(
 			ModelResourcePermissionRegistryUtil.getModelResourcePermission(
 				FileEntry.class.getName());
 
-		fileEntryModelResourcePermission.check(
-			permissionChecker, fileEntry.getFileEntryId(),
-			_getActionId(httpServletRequest));
+		try {
+			fileEntryModelResourcePermission.check(
+				permissionChecker, fileEntry.getFileEntryId(),
+				_getActionId(httpServletRequest));
+		}
+		catch (Exception exception) {
+			if (!AuthLoginGroupSettingsUtil.isPromptEnabled(
+					fileEntry.getGroupId())) {
+
+				throw new NoSuchFileEntryException(exception);
+			}
+
+			throw new PrincipalException(exception);
+		}
```

Comment:
Why catch a wider `Exception` exn? The `check` method throws a `PortalException`. Also, you should check if the exn is already a principal exception to avoid wrapping it in that case.